frontend/src/ui/components/HeroPreview/HeroPreview.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import './styles.scss';
+
+interface HeroPreviewProps {
+  homePageHeroTitleColor: undefined | string;
+  homePageHeroSubtitleColor: undefined | string;
+  homepageOverlayHtml: undefined | string;
+  heroCoverImage: undefined | string;
+}
+
+export const HeroPreview: React.FC<HeroPreviewProps> = (
+  props: HeroPreviewProps
+) => {
+  let titleStyles = {};
+  let subtitleStyles = {};
+  let containerStyles = {};
+  let matched;
+  const heroTextRegex = /^<h1>(.*)<\/h1><p>(.*)<\/p>$/;
+
+  if (props.homePageHeroTitleColor) {
+    titleStyles = {
+      color: props.homePageHeroTitleColor
+    };
+  }
+  if (props.homePageHeroSubtitleColor) {
+    subtitleStyles = {
+      color: props.homePageHeroSubtitleColor
+    };
+  }
+  if (props.homepageOverlayHtml) {
+    matched = heroTextRegex.exec(props.homepageOverlayHtml as string);
+  }
+  if (props.heroCoverImage) {
```

Comment:
Wouldn't it work without these checks? It seems to be falling back to defined styles, when these variables are `undefined`.

Reply:
@Agrendalath, the `.exec()` method requires a `string` and that is why the check for the existence of `props.homepageOverlayHtml` is required. The other 2 `if` conditions in this component check and set styles, so I don't think those can be removed.

Please let me know if I misunderstood your comment :)

Reply:
@lgp171188, I was referring to the `homePageHeroTitleColor`, `homePageHeroSubtitleColor`, but especially to `heroCoverImage`. CSS just falls back to another option if it receives something like `url("")`. Anyway, that's just a nit.

Reply:
> CSS just falls back to another option if it receives something like url(""). Anyway, that's just a nit.

@Agrendalath, I wasn't aware of that. But would prefer to keep it explicit :)

Reply:
@lgp171188, sure, makes sense!